File: app/src/main/java/com/udacity/sandwichclub/ui/adapter/SandwichAdapter.java
package com.udacity.sandwichclub.ui.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.udacity.sandwichclub.R;
import com.udacity.sandwichclub.model.Sandwich;

import java.util.List;

public class SandwichAdapter extends ArrayAdapter<Sandwich> {
    /* Class variables */
    private LayoutInflater inflater;
    private List<Sandwich> sandwiches;

    /* Class Constants */
    private static final String TAG = SandwichAdapter.class.getSimpleName();

    public SandwichAdapter(@NonNull Context context, List<Sandwich> sandwiches) {
        super(context, 0, sandwiches);
        inflater = LayoutInflater.from(context);
        this.sandwiches = sandwiches;
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        if (convertView == null) {
            convertView = inflater.inflate(R.layout.list_item_view, parent, false);
        }

        TextView sandwichQuickInfo = convertView.findViewById(R.id.sandwich_quick_info_tv);
        Sandwich sandwich = sandwiches.get(position);
        if (sandwich != null) {
            sandwichQuickInfo.setText(sandwich.getMainName());
            //todo: Affix circular image of sandwich
        } else {
            Log.e(TAG, "Sandwich at position " + position + " parsed as null");
        }

        return convertView;
    }
}

File: app/src/main/java/com/udacity/sandwichclub/ui/adapter/SandwichAdapter.kt
package com.udacity.sandwichclub.ui.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.navigation.findNavController
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.udacity.sandwichclub.R
import com.udacity.sandwichclub.model.Sandwich
import com.udacity.sandwichclub.ui.SandwichListFragmentDirections

class SandwichAdapter(private val sandwiches: List<Sandwich>) :
    RecyclerView.Adapter<SandwichAdapter.SandwichHolder>() {

    // companion object {
    //     val TAG = SandwichAdapter::class.simpleName
    // }

    class SandwichHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val image: ImageView = itemView.findViewById(R.id.sandwich_quick_info_iv)
        val name: TextView = itemView.findViewById(R.id.sandwich_quick_info_tv)
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SandwichHolder {
        val itemView =
            LayoutInflater.from(parent.context).inflate(R.layout.list_item_view, parent, false)
        return SandwichHolder(itemView)
    }

    override fun getItemCount(): Int {
        return sandwiches.size
    }

    override fun onBindViewHolder(holder: SandwichHolder, position: Int) {
        val sandwich = sandwiches[position]
        holder.name.text = sandwich.mainName

        val options = RequestOptions().error(R.drawable.ic_error).fitCenter().circleCrop()

        Glide.with(holder.image.context)
            .setDefaultRequestOptions(options)
            .load(sandwich.image)
            .into(holder.image)

        holder.itemView.setOnClickListener {
            val action =
                SandwichListFragmentDirections.actionSandwichListFragmentToSandwichFragment(
                    sandwich = sandwich,
                    fragmentLabel = sandwich.mainName
                )
            holder.itemView.findNavController().navigate(action)
        }
    }
}
